challenges/victor/backEndChallenge/routes/index.js

```diff
@@ -0,0 +1,39 @@
+var express = require("express");
+var router = express.Router();
+const { check, validationResult } = require("express-validator");
+
+var data;
+
+router.get("/data", (req, res) => {
+  res.json(data);
+});
+
+router.post(
+  "/data",
+  [
+    check("numbers")
+      .isArray({ min: 500, max: 500 })
+      .withMessage("You must submit a list of 500 numbers exactly."),
+    check("numbers")
+      .custom(
+        arr =>
+          !arr.some(index => {
+            !Number.isInteger(index);
+          })
+      )
+      .withMessage("only numbers can be submitted")
+  ],
+  (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(422).json({ errors: errors.array() });
+    }
+    data = req.body.numbers;
```

Comment:
What would happen if a `GET` request was submitted to the server that was processed before line 32 completed?

Reply:
If a GET request was sent before the POST request, The server would return undefined as the variable data has no value until the POST. If the GET request was sent after the POST request but before line 32 completed it would return the unsorted list of numbers. 
To resolve these I could start with dummy data in the data variable and move the sort function to the GET request function.